API to annotate block points

vlovich opened the pull request:
Fixes #44

emeryberger commented:
Please add some comments to the code, esp. regarding the new macros and their intended use case (ideally with an example), and I will merge.

emeryberger commented:
Bump @vlovich

vlovich commented:
Took another look and it turns out that I didn't actually implement the new `_coz_` C API so they'd always be no-ops. Fixed it up and added comments as best I could.
